src/Storage/Device.php

```diff
@@ -51,21 +52,26 @@ abstract public function getPath(string $filename): string;
      *
      * @param string $source
      * @param string $path
-     *
+     * @param int $chunk
+     * @param int $chunks
+     * @param string $tmp
+     * 
      * @throws \Exception
      *
-     * @return bool
+     * @return int
      */
-    abstract public function upload($source, $path): bool;
+    abstract public function upload($source, $path, $chunk = 1, $chunks = 1): int;
 
     /**
      * Read file by given path.
      *
      * @param string $path
+     * @param int $offset
+     * @param int $length
      *
      * @return string
      */
-    abstract public function read(string $path): string;
+    abstract public function read(string $path, int $offset = 0, int $length = null): string;
```

Comment:
It's not obvious why `(int) $offset` has a default value of 0, while `(int) $length` uses null. Can we find a pattern to be more consistent?

Reply:
@kodumbeats it's what the underlying `file_get_contents` accepts the params.

What do you suggest?

Reply:
That's a good enough reason for me :)